Bio/SeqIO/FastaIO.py

```diff
@@ -50,7 +54,22 @@ def SimpleFastaParser(handle):
     except StopIteration:
         return
     if not line.startswith(">"):
-        raise ValueError("FASTA files must start with '>'")
+        warnings.warn(
+            "Previously, the FASTA parser silently ignored comments at the\n"
+            "beginning of the FASTA file (before the first sequence).\n"
+            "\n"
+            "As the FASTA file format specification does not allow such\n"
+            "comments, this is now deprecated.\n"
+            "\n"
+            "In a future Biopython release, this deprecation warning will be\n"
+            "replaced by a ValueError. We encourage you to modify your FASTA\n"
+            "file to make it compliant with the FASTA file format\n"
+            "specification.",
+            BiopythonDeprecationWarning,
+        )
+        for line in handle:
+            if line.startswith(">"):
+                break
```

Comment:
I was thinking instead of a warning if there was a "header" and no sequences. e.g. Mistakenly parsing a non-FASTA file.

i.e. Historic usage of FASTA files with headers and at least one sequence would continue unaffected.

Reply:
Why? Following Biopython's deprecation policy, we remove code two releases and one year after a deprecation. That also breaks scripts. Why should we deviate from our policy in this particular case? Why, in this case, is it so important that scripts won't break?

Remember, this is a bug, not a feature. None of the other big software packages support such headers. Most software packages will report an error; Blast will accept the input file, but will try to interpret the header as an additional sequence. Biopython deviating from standards is asking for trouble.

I didn't find a description of the format with headers anywhere, not even in the Biopython documentation. We don't test for it either. The decision to allow such headers was taken on a blue Monday 25 years ago. See this comment in Tests/Fasta/f003:
```
# This file has comments in it
...
# I don't really know if FASTA format is supposed to support this
# but we might be able to without many problems
```
Are we now stuck with this for all eternity?

Yes, scripts may break, but as senior bioinformatics scientists, we also have an obligation to junior Biopython users to teach them to use sane methods. Deviating from established formats is not one of them. A meek, ignorable warning does not suffice.

For people who do want headers on their Fasta files, the only reasonable way to do so is to be consistent with Blast, and ignore lines starting with `;`, `#`, and `!`. But this should be an optional feature, maybe with an argument `comments=False|True`. People with Biopython-style headers in their Fasta files can then just put a `#` in front of each line in the header, and call `SeqIO.parse("myfile.fa", "fasta", comments=True)`.

Reply:
I disagree that this is a bug, but do accept that the mainstream consensus is not to use headers in FASTA files. This is a very handy mechanism for embedding metadata into FASTA files though (human readable headers or machine readable).